generator/template.go

```diff
@@ -232,15 +317,22 @@ func renderURL(r *registry.Registry) func(method data.Method) string {
 		url := method.URL
 		reg := regexp.MustCompile("{([^}]+)}")
 		matches := reg.FindAllStringSubmatch(url, -1)
+		fieldList := "["
 		if len(matches) > 0 {
 			log.Debugf("url matches %v", matches)
 			for _, m := range matches {
 				expToReplace := m[0]
 				fieldName := fieldNameFn(m[1])
 				part := fmt.Sprintf(`${req["%s"]}`, fieldName)
 				url = strings.ReplaceAll(url, expToReplace, part)
+				fieldList += fmt.Sprintf(`"%s", `, fieldName)
 			}
 		}
+		fieldList = strings.TrimRight(fieldList, ", ") + "]"
+
+		if !method.ClientStreaming && method.HTTPMethod == "GET" {
+			url += fmt.Sprintf("${fm.renderURLSearchParams(req, %s)}", fieldList)
```

Comment:
fieldList is not needed.

Reply:
`fieldList` is constructed here and used in `renderURLSearchParams` to remove the list of fields from the request object that is present in the URL path as parameters. This is because, `renderURLSearchParams` takes the whole request object as its input, which contains the key that may be present in the URL path as a parameter. So we need to ensure that fields in the request object that are present as URL path parameters are removed.

Reply:
`fieldList` as the name is quite confusing. a name like `fieldsInPath` will make the code reading better

Reply:
Thank you. I will change it as you suggested 🙇🏼 `fieldsInPath` makes more sense 👍🏼